Meter.c

```diff
@@ -120,40 +132,61 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
    assert(startPos <= w);
    assert(startPos + w <= RichString_sizeVal(bar));
 
+   const Settings* settings = this->host->settings;
    int blockSizes[10];
 
    // First draw in the bar[] buffer...
    int offset = 0;
+   const wchar_t* barChar = &bars[settings->barType][1];
+   const uint8_t barLen = wcslen(barChar);
+   const uint8_t wsub = w * barLen;
+
    for (uint8_t i = 0; i < this->curItems; i++) {
       double value = this->values[i];
+      int actualWidth = 0;
+
+      // ignore extremely small values
+      if((value / this->total) * wsub < 0.5){
+         blockSizes[i] = 0;
+         continue;
+      }
       if (isPositive(value) && this->total > 0.0) {
          value = MINIMUM(value, this->total);
+         actualWidth = ceil((value / this->total) * wsub);
          blockSizes[i] = ceil((value / this->total) * w);
       } else {
          blockSizes[i] = 0;
       }
       int nextOffset = offset + blockSizes[i];
       // (Control against invalid values)
       nextOffset = CLAMP(nextOffset, 0, w);
-      for (int j = offset; j < nextOffset; j++)
+
+
+      for (int j = offset; j < nextOffset; j++){
          if (RichString_getCharVal(bar, startPos + j) == ' ') {
             if (CRT_colorScheme == COLORSCHEME_MONOCHROME) {
                assert(i < strlen(BarMeterMode_characters));
                RichString_setChar(&bar, startPos + j, BarMeterMode_characters[i]);
+            } else if (settings->barType) {
+               RichString_setChar(&bar, startPos + j, bars[settings->barType][8]);
             } else {
                RichString_setChar(&bar, startPos + j, '|');
             }
          }
+      }
+
+      RichString_setChar(&bar, startPos + nextOffset-1, barChar[actualWidth % barLen]);
+
       offset = nextOffset;
    }
 
    // ...then print the buffer.
    offset = 0;
    for (uint8_t i = 0; i < this->curItems; i++) {
       int attr = this->curAttributes ? this->curAttributes[i] : Meter_attributes(this)[i];
-      RichString_setAttrn(&bar, CRT_colors[attr], startPos + offset, blockSizes[i]);
-      RichString_printoffnVal(bar, y, x + offset, startPos + offset, MINIMUM(blockSizes[i], w - offset));
-      offset += blockSizes[i];
+      RichString_setAttrn(&bar, CRT_colors[attr], startPos + offset, ceil(blockSizes[i]));
+      RichString_printoffnVal(bar, y, x + offset, startPos + offset, MINIMUM(ceil(blockSizes[i]), w - offset));
+      offset += ceil((double)blockSizes[i]);
```

Comment:
I see `ceil((double)blockSizes[i])` and `ceil(blockSizes[i])` and they look like two different calculations. `blocksizes` was defined as `int[10]` according to the code above, so the `ceil` function won't do what you want.

Reply:
oh yes you're right i had tried some other stuff thinking it would've been more readable forgot to remove that :+1: thanks for noticing :D